### ISSUE TITLE
Automatically run tests for more contributors

### ISSUE THREAD

**Issue by tarsius** [melpa/melpa#7965]
@milkypostman @purcell could one of you please go to https://github.com/melpa/melpa/settings/actions and set "Fork pull request workflows from outside collaborators" to "Require approval for first-time contributors who are new to GitHub". Thanks!

**Comment by milkypostman** [comment]
Can you say why we would want to do this? Is it just so that our tests run for people without approval?

**Comment by milkypostman** [comment]
I have made the change, but I have disabled any workflows that are not part of the "melpa" actions and reusable workflows. I've also disabled write access for actions. We don't use this so there should be no reason to have it enabled. Especially for a public repo like this one.

**Comment by tarsius** [comment]
> Is it just so that our tests run for people without approval?

Yes. This gives contributor the opportunity to fix an obvious error shortly after they open the faulty pr, instead of a few days later when someone gets around to clicking on "yes run the tests".

Limiting permissions to what we actually need also makes sense. :+1:

**Comment by tarsius** [comment]
@milkypostman You locked it down too much:

> actions/checkout@v2 is not allowed to be used in melpa/melpa. Actions in this workflow must be: within a repository owned by melpa.

**Comment by milkypostman** [comment]
Where are you seeing this? I enabled actions from melpa *and* those from GitHub. I wonder if that's enough.

**Comment by tarsius** [comment]
> Where are you seeing this?

Each and every test run: https://github.com/melpa/melpa/actions.

> I enabled actions from melpa and those from GitHub. I wonder if that's enough.

Should be I think but maybe you didn't save?

**Comment by milkypostman** [comment]
It looks like the most recent one ran. I don't know how to re-run the ones that didn't run properly.

**Comment by tarsius** [comment]
Normally the `...` menu allows doing that but that doesn't seem to be possible here.
